### DISCUSSION TITLE
Center view when at the bottom of the buffer

### DISCUSSION THREAD

**Discussion by smartmic** [orbitalquark/textadept#36] [Q&A] [answered]
Hi there,

I just discovered textadept and I am still finding my way around.  Very often, when I am at the bottom of both the buffer and window, I want to (vertically) center my current line.  So I figured out to assign a key binding but it only works reliable if my cursor is not at the bottom (window) half of the buffer.  Eg. if I am on the last line of a long text buffer, hit my key binding and start editing, the cursor line jumps to the window bottom again.  

How can I keep the vertical center also for editing?

I used this key binding assignment:

    keys['alt+c'] = function() _G.view.vertical_center_caret() end

Any help is welcome — thanks in advance!

Martin

**Comment by orbitalquark** [comment, marked as answer]
In your *~/.textadept/init.lua*:

    view.end_at_last_line = false

See https://orbitalquark.github.io/textadept/api.html#view.end_at_last_line
